dipy/tracking/streamlinespeed.pyx

```diff
@@ -102,15 +102,18 @@ def length(streamlines):
         arclengths = np.zeros(len(streamlines), dtype=np.float64)
 
         if streamlines.data.dtype == np.float32:
-            c_arclengths_from_arraysequence[float2d](streamlines.data,
-                                                     streamlines._offsets,
```

Comment:
Streamlines is a nibabel.streamlines.ArraySequence object! So, I guess this is a question to the nibabel developers :-)

More seriously, from: https://github.com/nipy/nibabel/blob/master/nibabel/streamlines/array_sequence.py#L77-L78 it does look like that's a np.intp! Changed that cast to intp instead.